RecoLocalTracker/SubCollectionProducers/src/ClusterMultiplicityFilter.cc

```diff
@@ -15,19 +15,15 @@
 #include <memory>
 
 #include "RecoLocalTracker/SubCollectionProducers/interface/ClusterMultiplicityFilter.h"
-
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-
-#include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
-#include "DataFormats/Common/interface/DetSetVector.h"
-
 #include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
 
 
-ClusterMultiplicityFilter::ClusterMultiplicityFilter(const edm::ParameterSet& iConfig)
+ClusterMultiplicityFilter::ClusterMultiplicityFilter(const edm::ParameterSet& iConfig) :
+  maxNumberOfClusters_(iConfig.getParameter<unsigned int>("MaxNumberOfClusters")),
+  clusterCollectionTag_(iConfig.getParameter<edm::InputTag>("ClusterCollection"))
 {
-  maxNumberOfClusters_    = iConfig.getUntrackedParameter<unsigned int>("MaxNumberOfClusters");
-  clusterCollectionLabel_ = iConfig.getUntrackedParameter<std::string>("ClusterCollectionLabel");
+  clusters_ = consumes<edmNew::DetSetVector<SiStripCluster> >(clusterCollectionTag_);
```

Comment:
You could make this const as well if you move its initialization to before the ```{```.